docs/architecture/vaults/overview.puml

```diff
@@ -0,0 +1,88 @@
+@startuml
+hide empty members
+
+package misc {
+    abstract Multicall
+    abstract ReentrancyProtection
+    abstract Recoverable
+}
+
+package common {
+    class Gateway
+    class MessageDispatcher
+}
+
+package external {
+    class ERC20
+    class ERC6909
+}
+
+class PoolManager
+
+class AsyncRequestManager
+class SyncRequestManager
+
+class AsyncVault
+class SyncDepositVault
+
+class Escrow
+class PoolEscrow
+
+class ShareToken
+
+class BalanceSheet
+class VaultRouter
+
+interface IVaultFactory
+abstract BaseVault
+
+class AsyncVaultFactory
+class PoolEscrowFactory
+class SyncDepositVaultFactory
+class TokenFactory
+
+AsyncVaultFactory --|> IVaultFactory
+SyncDepositVaultFactory --|> IVaultFactory
+
+PoolManager --up-> TokenFactory
+PoolManager --up-> PoolEscrowFactory
+PoolManager --up->"n" IVaultFactory
+PoolManager <-left-> BalanceSheet
+PoolManager --down->"n" ShareToken
+PoolManager ---down->"n" BaseVault
+PoolManager --> Gateway
+PoolManager --> MessageDispatcher
+
+BalanceSheet -up--> PoolEscrowFactory
+BalanceSheet --> MessageDispatcher
+
+SyncRequestManager <-down->"n" SyncDepositVault
+SyncRequestManager -up-> PoolManager
+SyncRequestManager -up-> PoolEscrowFactory
+SyncRequestManager -up-> BalanceSheet
+SyncDepositVault --|> BaseVault
+
+AsyncRequestManager <-down->"n" AsyncVault
+AsyncRequestManager <-down->"n" SyncDepositVault
+AsyncRequestManager -up-> PoolManager
+AsyncRequestManager -up-> BalanceSheet
+AsyncRequestManager -up-> PoolEscrowFactory
+AsyncRequestManager --> MessageDispatcher
+
+VaultRouter -left-> PoolManager
+VaultRouter -right-> Escrow
+
+PoolEscrowFactory -up-> PoolEscrow
+PoolEscrow --|> Escrow
+
+BaseVault --> ShareToken
+BaseVault --> ERC20
+BaseVault --> ERC6909
```

Comment:
CR: Remove.

Sadly, we had to remove ERC6909 support for vaults in #309. IIRC the main reason was that the ERC4626 interface for the SyncDepositVault does not support it.

Reply:
Done. Updated diagram